answers.md

```diff
@@ -0,0 +1,50 @@
+# Answers
+## HPC good citizenship
+1. No. `-pe openmp 64` only tells the grid engine to allocate 64 cores. But the real number of processors used are determined by the programs running on it.
+
+
+2. Review the scripts before submitting to the grid engine. Softwares written by others usually have documentations. If not, try running the softwares with a smaller dataset as a test.
+
+
+3. Use following command
+
+```
+/usr/bin/time -f "maximum resident set size = %M, elapsed real time = %E, cpu time = %S, user time = %U" command
+```
+The idea is from this [page](https://unix.stackexchange.com/questions/375889/unix-command-to-tell-how-much-ram-was-used-during-program-runtime).
```

Comment:
this is good but it is is also good to add option -o which allows an output file. env time -v -o MYOUTPUT CMD would do the trick and do a dump to a text file. this helps sometimes if you care about what goes in your stdout and stderr files. I like to keep it seperate as I usually do tail -n 1 MYSTDOUTFILE to see if I was able to run to completion and then sort my STDERRFILES using ls -lS jobname.ejobid* by size to see if the stderr produced larger output than normal. this typically means there where errors on the run. This doesn't work though when the programmers are sloppy and lazy and output stdout to stderr. :(